packages/hooks_riverpod/lib/src/use_provider_listener.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import 'internals.dart';
+
+/// A hook that listens to a provider and and triggers a callback whenevr the value of provider is changed.
+///
+/// As a usage example, consider:
+///
+/// ```dart
+/// class Counter extends StateNotifier<int> {
+///   Counter(): super(0);
+///    void increment() => state++;
+/// }
+///
+/// final counterProvider = StateNotifierProvider((ref) => Counter());
+/// ```
+///
+/// We can then use [useProviderListener] to listen to `helloWorldProvider` like so:
```

Comment:
What is `helloWorldProvider` ?
Do you mean `counterProvider` ?

Reply:
Yeah good catch I'll correct that